phoenix-core/src/main/java/org/apache/phoenix/compile/UpsertCompiler.java

```diff
@@ -804,6 +804,7 @@ public Long getEstimatedBytesToScan() throws SQLException {
             ////////////////////////////////////////////////////////////////////
             // UPSERT SELECT run client-side
             /////////////////////////////////////////////////////////////////////
+            ScanUtil.setClientSideUpsertSelect(queryPlan.getContext().getScan());
```

Comment:
This would be better to encapsulate in the StatementPlan if we even need it. Why do we need it?